roofit/histfactory/src/HistoToWorkspaceFactory.cxx

```diff
@@ -1005,6 +1005,11 @@ namespace HistFactory{
       poi = (RooRealVar*) params_obj;
       cout << "printing results for " << poi->GetName() << " at " << poi->getVal()<< " high " << poi->getErrorLo() << " low " << poi->getErrorHi()<<endl;
     }
+    delete params_itr;
+    if (!poi) {
+       cerr << "found no POI" << endl;
```

Comment:
This is also a case for the RooMsgService

Reply:
Should I fix myself or you do it later?

Reply:
I will fix it later. There is much work to be done with histfactory anyway.